### DISCUSSION TITLE
Trouble with strong name

### DISCUSSION THREAD

**Discussion by bzuidgeest** [dnSpyEx/dnSpy#152] [General]
Hello,

I have an dll that is loaded by an application like a plugin. Unfortunately the plugin dll has a bug. It is also no longer maintained (By Microsoft) and I don't have the code.

I found dnSpy and can load the assembly, I make the change (adding a single substring call to a function) and I save it. 

The program (mim2016) refuses to load the new patched assembly with an error in eventlog

  "Could not load file or assembly 'Microsoft.Azure.ActiveDirectory.Connector, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. Strong name signature could not be verified.  The assembly may have been tampered with, or it was delay signed but not fully signed with the correct private key. (Exception from HRESULT: 0x80131045)

It was certainly tampered with -> by me. I am no hacker. Any advise/tips out there for how I can get the application to load the assembly with modifications.

I tried googling it, but the information confuses me greatly. Should I just remove the SN (snremove)? I tried somthing with SN.exe -Vr but that changed little.

**Comment by UlyssesWu** [comment]
`SN.exe -Vr {path to your dll}` should work.

**Reply by bzuidgeest** [reply]
Thank you for replying.

I have already tried that but it didn't change anything. Still the same error. I think it might be possible the application hosting the plugin is explicitly asking for the dll by its strong name or something....

When I have a change I can retry that.

**Reply by GazziFX** [reply]
I think you can remove strong name in base application

**Reply by UlyssesWu** [reply]
Check if your dll is in GAC. If it is, remove it from GAC and try again.

**Comment by ElektroKill** [comment]
In order for the file to load properly and not result in the `BadImageFormatException` being thrown it will need to be resigned and will need to have a fully valid strong name signature.

**Comment by mitchcapper** [comment]
This is an old thread, but just incase someone else runs into this.  You might need to have a dll that has a strong name but modern net (3.5  SP1 and higher) can be configured (and is by default) to not verify many signatures as long as the files are in the right place.   https://learn.microsoft.com/en-us/dotnet/standard/assembly/disable-strong-name-bypass-feature talks about it a bit.
